src/components/course/tests/CourseSidebarPrice.test.jsx

```diff
@@ -73,90 +77,122 @@ const courseStateWithNoOffersNoLicenseSubsidy = {
 );
 
 SidebarWithContext.propTypes = {
   initialAppState: PropTypes.shape({}).isRequired,
   initialCourseState: PropTypes.shape({}).isRequired,
   userSubsidyState: PropTypes.shape({}).isRequired,
+  subsidyRequestsState: PropTypes.shape({}).isRequired,
 };
 
 const SPONSORED_BY_TEXT = 'Sponsored by test-enterprise';
 
-describe('Sidebar price display with hideCourseOriginalPrice ON, No subsidies', () => {
-  test('no subsidies, shows original price, no messages', () => {
-    render(<SidebarWithContext initialCourseState={courseStateWithNoOffersNoLicenseSubsidy} />);
-    expect(screen.getByText(/\$7.50 USD/)).toBeInTheDocument();
-    expect(screen.queryByText(INCLUDED_IN_SUBSCRIPTION_MESSAGE)).not.toBeInTheDocument();
-    expect(screen.queryByText(SPONSORED_BY_TEXT)).not.toBeInTheDocument();
+describe('<CourseSidebarPrice/> ', () => {
+  describe('Browse and Request', () => {
+    test('Display correct message when browse and request on and no subsidy', () => {
+      config.features.FEATURE_BROWSE_AND_REQUEST = true;
+      render(
+        <SidebarWithContext
+          initialCourseState={courseStateWithNoOffersNoLicenseSubsidy}
+          subsidyRequestsState={{
+            ...defaultSubsidyRequestsState,
+            subsidyRequestConfiguration: { subsidyRequestsEnabled: true },
+          }}
+        />,
+      );
+      expect(screen.getByText(/\$7.50 USD/)).toBeInTheDocument();
+      expect(screen.getByTestId('browse-and-request-pricing')).toBeInTheDocument();
+      expect(screen.queryByText(INCLUDED_IN_SUBSCRIPTION_MESSAGE)).not.toBeInTheDocument();
+      expect(screen.queryByText(SPONSORED_BY_TEXT)).not.toBeInTheDocument();
+    });
   });
-});
 
-describe('Sidebar price display with hideCourseOriginalPrice ON', () => {
-  test('subscription license subsidy, shows no price, correct message', () => {
-    render(<SidebarWithContext initialCourseState={courseStateWithLicenseSubsidy} />);
-    expect(screen.queryByText(/\$7.50 USD/)).not.toBeInTheDocument();
-    expect(screen.getByText(INCLUDED_IN_SUBSCRIPTION_MESSAGE)).toBeInTheDocument();
-    expect(screen.queryByText(SPONSORED_BY_TEXT)).not.toBeInTheDocument();
-  });
-  test('offer 100% subsidy, shows no price, correct message', () => {
-    render(<SidebarWithContext
-      initialCourseState={courseStateFullOfferSubsidy}
-    />);
-    expect(screen.queryByText(/\$7.50 USD/)).not.toBeInTheDocument();
-    expect(screen.queryByText(/\$0.00 USD/)).not.toBeInTheDocument();
-    expect(screen.queryByText(INCLUDED_IN_SUBSCRIPTION_MESSAGE)).not.toBeInTheDocument();
-    expect(screen.getByText(SPONSORED_BY_TEXT)).toBeInTheDocument();
+  describe('Sidebar price display with hideCourseOriginalPrice ON, No subsidies', () => {
+    test('no subsidies, shows original price, no messages', () => {
+      render(<SidebarWithContext initialCourseState={courseStateWithNoOffersNoLicenseSubsidy} />);
+      expect(screen.getByText(/\$7.50 USD/)).toBeInTheDocument();
```

Comment:
nit: you don't have to call toBeInTheDocument() when using getByText since it will throw an error if the element is not found